packages/lit-html/src/directive-helpers.ts

```diff
@@ -121,7 +121,12 @@ export const insertPart = (
     );
   } else {
     const endNode = wrap(part._$endNode!).nextSibling;
-    if (endNode !== refNode) {
+    const parentChanged = part._$parent !== containerPart;
+    if (parentChanged) {
+      part._$reparentDisconnectables?.(containerPart);
+      part._$parent = containerPart;
```

Comment:
Setting `_$parent` appears to be unnecessary as it's done in `_$reparentDisconnectables`?

Reply:
Mentioned above, because `_$reparentDisconnectables` is patched in, we can't rely on that to set `_$parent`.

`_$reparentDisconnectables` does set `_$parent` in between unlinking/re-linking because the API s were set to work off the current state of the part; I tried refactoring `removeDisconnectableFromParent` and `addDisconnectableToParent` but it felt weird to ignore the initial part's `_$parent` state and use an argument, but use subsequent part's `_$parent` as we climb up the tree.

I'll add a comment for now; lmk if you think we should refine that.

Reply:
Done.